### DISCUSSION TITLE
What are all the measures taken to avoid spammers in the NPM registry?

### DISCUSSION THREAD

**Discussion by Thanga-Ganapathy** [npm/feedback#1112] [Registry]
No description needed just go through the following links you will know what it is.

https://www.npmjs.com/~shivamkalsi2024

May I know is there any package keywords limit is there? because these type of pkgs polluting the registry.

Try tihs: https://www.npmjs.com/search?q=object%20diff&ranking=maintenance&page=2&perPage=20

I already commented a similar one here: (No response yet)

https://github.com/npm/feedback/discussions/856

I just found randomly but who knows and how many are them?

Please take necessary actions ASAP.

**Comment by Thanga-Ganapathy** [comment]
Please have a look at the keywords list

![image](https://github.com/npm/feedback/assets/22178434/0ca28734-f8d7-47b0-8572-8aaa9c604236)

**Comment by Thanga-Ganapathy** [comment]
![image](https://github.com/npm/feedback/assets/22178434/0d640648-ec43-4665-8b08-17743b724f13)

Please have look at the `index.js` code, dependencies within themselves, weekly downloads and the final effect on the popularity scores. 

Which means there is another issue lined up for you.

FYI: https://github.com/npm/feedback/discussions/1104

**Comment by Thanga-Ganapathy** [comment]
![image](https://github.com/npm/feedback/assets/22178434/cbb4bde0-aa6d-4650-bb66-12a82b1872a3)

**Comment by leobalter** [comment]
Thanks for the report. Moving forward you can use our [support forms](https://www.npmjs.com/support) to report spam.

**Reply by Thanga-Ganapathy** [reply]
Hello @leobalter this is not just spam report, but I asked you a question, please read the title carefully.

And I don't want to simply go through your form to fill out the spam findings.

Please reopen this.

**Reply by leobalter** [reply]
Hi @Thanga-Ganapathy! This is all I can share at the moment. Although, I'll see how we can expand our public documentation regarding measures against spam.
